scripts/research/lib/batch02_evidence_retention.py

```
@@ -0,0 +1,1036 @@
+            )
+            head = run_git(isolated, ["rev-parse", "HEAD"]).decode("utf-8").strip().lower()
+            run_git(isolated, ["push", "origin", f"HEAD:{evidence_ref}"])
+            expected_head = head
+        else:
+            run_git(isolated, ["fetch", "--no-tags", "origin", evidence_ref])
+            run_git(isolated, ["checkout", "--detach", "FETCH_HEAD"])
+            existing_bytes = (isolated / RESERVATION_BLOB_PATH).read_bytes()
+            if existing_bytes != payload_bytes:
+                raise PreOutcomeRetentionError(
+                    "incompatible pre-existing evidence reservation"
+                )
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)